app/code/Magento/Captcha/Model/Image.php

```diff
@@ -0,0 +1,1136 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Captcha\Model;
+
+use Exception;
+use InvalidArgumentException;
+use Laminas\Math\Rand;
+use Laminas\Session\Container;
+use Laminas\Stdlib\ErrorHandler;
+use Laminas\Validator\AbstractValidator;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Math\Random;
+use Traversable;
+
+class Image extends AbstractValidator
```

Comment:
Adding such a big amount of new code to Magento core instead of re-using some existing components from Laminas isn't a good idea. Magento definitely has a slower release cycle and even doesn't have test coverage for this functionality.
Fixing issues in this area will be not as fast as just updating laminas component.
Can we just contribute fixes to Laminas Captcha, instead of re-inventing wheel?
/CC @sivaschenko